spec/features/search_page_spec.rb

```diff
@@ -4,63 +4,40 @@
 
 RSpec.feature 'Search page', type: :feature, elasticsearch: true do
   scenario 'Displays a not found message when a search returns 0 results' do
-    query = 'interesting dataset'
-
-    search_for(query)
-
+    search_for('interesting dataset')
     expect(page).to have_css('h1', text: 'Search results')
     expect(page).to have_content("0 results found")
   end
 
   scenario 'Displays search results' do
-    dataset_title = 'A very interesting dataset'
-    query = 'interesting dataset'
-
-    dataset = DatasetBuilder.new
-                .with_title(dataset_title)
-                .build
-
+    dataset = build :dataset, title: 'A very interesting dataset'
     index(dataset)
-    search_for(query)
+    search_for('interesting dataset')
 
     expect(page).to have_css('h1', text: 'Search results')
-    expect(page).to have_css('a', text: dataset_title)
+    expect(page).to have_css('a', text: dataset.title)
   end
 
   scenario 'Search results are correctly sorted' do
-    old_dataset = DatasetBuilder.new
-                    .with_title('Old Interesting Dataset')
-                    .with_name('old-dataset')
-                    .public_updated_at(Time.now - 1)
-                    .build
-
-    new_dataset = DatasetBuilder.new
-                    .with_title('Recent Interesting Dataset')
-                    .with_name('new-dataset')
-                    .build
-
+    old_dataset = build :dataset, title: 'Old Interesting Dataset', public_updated_at: 1.hour.ago
+    new_dataset = build :dataset, title: 'Recent Interesting Dataset'
     index(old_dataset, new_dataset)
 
     search_for('Old Interesting Dataset')
-
     expect(page).to have_css('option[selected]', text: 'Best match')
```

Comment:
The large number of new lines here was making the test hard to follow - removed a few to group the together a bit more.